test/e2e/framework/prometheus.go

```diff
@@ -168,3 +170,53 @@ func (f *Framework) DeletePrometheusAndWaitUntilGone(name string) error {
 func promImage(version string) string {
 	return fmt.Sprintf("quay.io/prometheus/prometheus:%s", version)
 }
+
+func (f *Framework) WaitForTargets(amount int) error {
+	var targets []*Target
+
+	if err := f.Poll(time.Minute*10, time.Second, func() (bool, error) {
+		var err error
+		targets, err = f.GetActiveTargets()
+		if err != nil {
+			return false, err
+		}
+
+		if len(targets) == amount {
+			return true, nil
+		}
+
+		return false, nil
+	}); err != nil {
+		return fmt.Errorf("Waiting for targets timed out. %v of %v targets found. %v", len(targets), amount, err)
```

Comment:
> Error strings should not start with a capital letter because they'll often be prefixed before printing

https://github.com/golang/go/wiki/Errors#errors